### DISCUSSION TITLE
How to pause or stop when running Play bag

### DISCUSSION THREAD

**Discussion by lishulincug** [foxglove/community#721] [General]
How to pause when running Play bag。
await bag.readMessages(
                    // { topics: ["/camera_array/cam0/image_raw/theora"] },
                    {
                        topics: selectTopic
                    },(result) => {
console.log('result',result)
})

**Comment by jtbandes** [comment]
Use the `bag.messageIterator` method instead + `for await`, see example here: https://github.com/foxglove/rosbag#quick-start
